pkg/controller/operands/networkAddons.go

```diff
@@ -61,12 +61,24 @@ func (h *cnaHooks) updateCr(req *common.HcoRequest, Client client.Client, exists
 		return false, false, errors.New("can't convert to CNA")
 	}
 
+	// If deployOVS annotation doesn't exists prior the upgrade - set this annotation to true;
+	// Otherwise - remain the value as it is.
+	if req.UpgradeMode {
```

Comment:
let me know if the code already does this, but we should *not* do this on each and every upgrade. We should add the annotation only if we upgrade from an "old" cnao, e.g. < v0.43.0.

Reply:
This code adds the annotation if `UpgradeMode==true` AND the annotation is not already there.
This is agnostic of the version, but notice that after the first upgrade, the annotation will be there so this code will not change the annotation.

Reply:
@RamLavi assume someone has a fresh install of cnao v0.43.0 (without ovs). Nothing adds any annotation to HCO.
What would happen when they upgrade to cnao v0.44.0 in 3 months?

Reply:
CNAO is not used standalone in CNV context, but controlled by HCO. Let's talk in HCO terms.
- when someone installs a fresh install of HCO, he will use the alm-example and gets the `deployOvs:false` annotation that will cause CNAO to not deploy ovs. when he upgrades in 3 months, the ovs will not be deployed, due to this annotation.
- when someone installs a fresh install of HCO with `deployOvs:true`, ovs will be deployed and this will persist to future upgrades.
- (your case) when someone installs a fresh install of HCO without adding the annotation, ovs will not e deployed, but on upgrade it'll be added `deployOvs:true` and ovs will be deployed. Is this a valid and supported scenario though?

Reply:
> he will use the alm-example and gets the `deployOvs:false` annotation

This is the bit that I was missing.

> (your case) when someone installs a fresh install of HCO without adding the annotation, ovs will not e deployed, but on upgrade it'll be added `deployOvs:true` and ovs will be deployed. Is this a valid and supported scenario though?

Yes, it is supported and [documented](https://docs.openshift.com/container-platform/4.6/virt/install/installing-virt-cli.html#virt-subscribing-cli_installing-virt-cli). We must not assume that we are installed via GUI.

Reply:
I updated the PR.
On upgrade, I'm checking whether OVS exists on CNAO CR if the `deployOVS` annotation does not exist.

- In case ovs exists - the annotation is being set to true (and therefore CNAO CR will be reconciled with `ovs: {}`).
- In case ovs doesn't exist - the annotation is being set to false (and therefore CNAO CR will be reconciled _without_ the ovs field).

That way we preserve the existence / in-existence of OVS between upgrades.

Reply:
reviewing the HLD, I'm afraid I wasn't accurate in my last case. let me clear it up:
- (your case) when someone installs a fresh install of HCO without adding the annotation, ovs will not be deployed, **and the `deployOvs:false` annotation will be added** . When upgrading, the annotation will keep the ovs from being further deployed.

@dankenigsberg sorry for the mistake.
@orenc1 I don't think we actually need to check the CNAO-CR. in previous versions, the `deployOVS` annotation does not exist, and ovs is always deployed. on the new version, the annotation is always added (as mentioned before - if not exist, will be reconciled to add `deployOvs:false` as this is the default behavior unless we're on upgrade mode)

Reply:
done.

Reply:
@dankenigsberg FYI we updated the [design](https://docs.google.com/document/d/1-6hdXgaWF3RNZM-6dw5mPn4r78SmtTiTnfffldHr-1Q/edit)